### ISSUE TITLE
Dialog de Configuracoes solicitado eternamente

### ISSUE THREAD

**Issue by luciogj** [stone-payments/sdk-ios-v2#13]
o metodo validatePinpadConnection retorna NO mesmo estando pareado o pinpad com o device.

**Comment by luciogj** [comment]
Error Domain=br.com.stone.StoneSDK Code=303 "Connection to Pinpad not found." UserInfo={NSLocalizedDescription=Connection to Pinpad not found.}

**Comment by jaisonv** [comment]
@luciogj, depois de parear o pinpad, foi estabelecida uma sessão com o mesmo utilizando o provider STNPinPadConnectionProvider?

Essa parte da documentação explica sobre isso: [Objective-C](https://github.com/stone-payments/sdk-ios-v2/tree/master/objc#criação-de-sessão-com-o-pinpad) / [Swift](https://github.com/stone-payments/sdk-ios-v2/tree/master/swift#criação-de-sessão-com-o-pinpad)

**Comment by luciogj** [comment]
Sim foi feito.
Em relação a versão do sdk 2.0.1 para a mais recente 2.1.2 houve alguma mudança de procedimento?
pergunto isso porque o mesmo app com o sdk 2.0.1 esta conectando.

**Comment by jaisonv** [comment]
Sim. Fizemos bastante alterações quando passamos para a versão 2.1.0. Pode conferir como está sendo feito em nossa demo.

**Comment by luciogj** [comment]
Desculpe, estava me referindo somente ao procedimento de conexão ao pinpad.
voltando ao primeiro comentario, ao executar o metodo validatePinpadConnection retorna NO mesmo estando pareado o pinpad com o device. Chamando como esta na documentação.

**Comment by luciogj** [comment]
o metodo STNPinPadConnectionProvider deve ser executado após, no entanto o método validatePinPadConnection no SDK mais recente retorna NO (false).

**Comment by luciogj** [comment]
Na tentativa de execução do método sendTransaction obtive o seguinte retorno: Error Domain=br.com.stone.StoneSDK Code=203 "Invalid value for transaction." UserInfo={NSLocalizedDescription=Invalid value for transaction.} 
Dados da transação: 

   STNTransactionModel *transaction = [[STNTransactionModel alloc] init];
        transaction.amount = [NSNumber numberWithInt: ];
        transaction.initiatorTransactionKey = idConta;
        
        if (operationType == DEBITO) {
            transaction.type = STNTransactionTypeSimplifiedDebit;               // transação no débito
            transaction.instalmentType = STNInstalmentTypeNone;                 // tipo de parcelamento: nenhum
            transaction.instalmentAmount = STNTransactionInstalmentAmountOne;   // número de parcelas: 1
        } else {
            transaction.type = STNTransactionTypeSimplifiedCredit;                 // transação no credito
            transaction.instalmentType = STNInstalmentTypeMerchant;                // tipo de parcelamento: s/juros (c/ adquirente)
            transaction.instalmentAmount = [self valueOfInstalment :instalment];   // número de parcelas
        }

**Comment by luciogj** [comment]
o valor (amount) setado como 10 a transacao passou no entanto não abriu no pinpad a senha a ser digitada.

**Comment by jaisonv** [comment]
Pode testar com essa versão?

[StoneSDK_v2.1.3.zip](https://github.com/stone-payments/sdk-ios-v2/files/1122437/StoneSDK_v2.1.3.zip)

**Comment by luciogj** [comment]
ok. testado. abaixo o retorno de erro.
2017-07-04 20:26:59.570 StarPay[4857:1394362] Error Domain=br.com.stone.StoneSDK Code=203 "Invalid value for transaction." UserInfo={NSLocalizedDescription=Invalid value for transaction.}

**Comment by jaisonv** [comment]
Ah, ok. Nesse caso o erro ocorre por causa do valor. Passa 1 real no mínimo.

**Comment by luciogj** [comment]
Jaison nas versões anteriores do sdk são permitidas transações inferiores a 1 real. Vocês incluiram essa restrição?

**Comment by jaisonv** [comment]
Sim.

**Comment by luciogj** [comment]
a despeito disso, as transações acima de 1 real também deram problema

**Comment by luciogj** [comment]
Jaison, vamos ficar nesse dialogo eternamente, ja solicitei uma call com a equipe de vocês para tentarmos resolver o problema em minutos e não em dias como estamos fazendo aqui.

**Comment by luciogj** [comment]
Infelizmente por aqui nao estamos sendo produtivos para resolver um problema que me parece simples.

**Comment by jaisonv** [comment]
Me passa seu skype.

**Comment by luciogj** [comment]
lucio.juliano
